File: src/main/java/ru/translater/translate/TranslateProperties.java
package ru.translater.translate;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import ru.translater.common.CommonProperties;
import ru.translater.translate.yandex.YandexConnectionProperties;

@Data
@Configuration
@ConfigurationProperties(prefix = "translate")
public class TranslateProperties extends CommonProperties {
    private String analyzePath;
    private YandexConnectionProperties yandexConnectionProperties;
    private String configPath;
}

File: src/main/kotlin/com/translater/translate/TranslateProperties.kt
package com.translater.translate

import com.translater.translate.yandex.properties.YandexPartConnectionProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "translate")
class TranslateProperties(
    val yandex: YandexPartConnectionProperties,
    val storePath: String,
    val analyzePath: String,
    val configPath: String,
    val sourceLanguage: String
)